secret/cert/cert.go

```diff
@@ -0,0 +1,144 @@
+package cert
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/giantswarm/certificatetpr"
+	microerror "github.com/giantswarm/microkit/error"
+	micrologger "github.com/giantswarm/microkit/logger"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/pkg/api"
+	apierrs "k8s.io/client-go/pkg/api/errors"
+	"k8s.io/client-go/pkg/api/v1"
+	"k8s.io/client-go/pkg/watch"
+)
+
+const (
+	// WatchTimeOut is the time to wait on watches against the Kubernetes API
+	// before giving up and throwing an error.
+	WatchTimeOut = 90 * time.Second
+)
+
+// Config represents the configuration used to create a service.
+type Config struct {
+	// Dependencies.
+	K8sClient kubernetes.Interface
+	Logger    micrologger.Logger
+}
+
+// DefaultConfig provides a default configuration to create a new service by
+// best effort.
+func DefaultConfig() Config {
+	return Config{
+		// Dependencies.
+		K8sClient: nil,
+		Logger:    nil,
+	}
+}
+
+// New creates a new configured service.
+func New(config Config) (*Service, error) {
+	// Dependencies.
+	if config.K8sClient == nil {
+		return nil, microerror.MaskAnyf(invalidConfigError, "config.K8sClient must not be empty")
+	}
+	if config.Logger == nil {
+		return nil, microerror.MaskAnyf(invalidConfigError, "config.Logger must not be empty")
+	}
+
+	newService := &Service{
+		// Dependencies.
+		k8sClient: config.K8sClient,
+		logger:    config.Logger,
+	}
+
+	return newService, nil
+}
+
+// Service implements the version service interface.
+type Service struct {
+	// Dependencies.
+	k8sClient kubernetes.Interface
+	logger    micrologger.Logger
+}
+
+// SearchCerts watches for all secrets of a cluster  and returns it as
+// certificate bundle.
```

Comment:
It's a minor semantic issue, but the bundle doesn't only contain certs, but certs and keys.